builtin/logical/consul/path_roles.go

```diff
@@ -29,13 +29,30 @@ func pathRoles(b *backend) *framework.Path {
 				Description: "Name of the role.",
 			},
 
+			// The "policy" and "token_type" parameters were deprecated in Consul back in version 1.4.
+			// They have been removed from Consul as of version 1.11. Consider removing them here in the future.
 			"policy": {
 				Type: framework.TypeString,
 				Description: `Policy document, base64 encoded. Required
 for 'client' tokens. Required for Consul pre-1.4.`,
+				Deprecated: true,
+			},
+
+			"token_type": {
+				Type:    framework.TypeString,
+				Default: "client",
+				Description: `Which type of token to create: 'client' or 'management'. If
+a 'management' token, the "policy", "policies", and "consul_roles" parameters are not
+required. Defaults to 'client'.`,
+				Deprecated: true,
 			},
 
 			"policies": {
+				Type:        framework.TypeCommaStringSlice,
+				Description: `Use "consul_policies" instead.`,
```

Comment:
nit: Wondering if we can provide more detail in the description here. Something that indicates that `policies` still works but is replaced by `consul_policies` 🤔 We may also want to mention that `consul_policies` has precedence over `policies` (i.e., if `consul_policies` used, `policies` will not be used).

This will also be relevant for the API doc updates.

Reply:
I'm totally open to it. I wasn't sure if this was an appropriate place for a full explanation or if it should stay in the docs, so I followed the format the lease field used.

Reply:
Gotcha! I didn't know that format was used elsewhere. I think the details I mentioned can be saved for the API docs then.

Reply:
Sounds like a plan. I agree it would be good to specifically mention the precedence consul_policies has.

Reply:
I just stumbled upon this https://github.com/hashicorp/vault/blob/main/sdk/helper/tokenutil/tokenutil.go#L274-L276. Too bad it's scoped to tokenutil. I think it could be more broadly useful for consistency.

Reply:
That's an interesting find. It also reminds me that me that Jim and I spoke about a desire for warnings outputted on use of deprecated features. Vault can already acknowledge unrecognized parameters and gives a warning for them, so following that design would feel consistent. Just thought I'd spread the idea around.

Example:
```
WARNING! The following warnings were returned from Vault:

  * Endpoint ignored these unrecognized parameters: [test_param]
```